### PR TITLE
Add joint state publisher during simulation

### PR DESCRIPTION
**Pull request by egordon**
Merge after #56 

This is a useful feature to get the current joint states (as reported by the MetaSkeleton) even if we are running in simulation using the same ROS topic interface as for the real robot.

Tested on the simulated feeding demo (ada_demos master as of right now).

***

**Before creating a pull request**

- [ N/A ] Document new methods and classes
- [ ] Format code with `make format`

**Before merging a pull request**

- [ N/A, see above ] Add unit test(s) for this change

### CODE REVIEW

File: src/Ada.cpp [hunk 1]
```diff
@@ -363,6 +367,26 @@ ConstAdaHandPtr Ada::getHand() const
 void Ada::update()
 {
   step(std::chrono::system_clock::now());
+
+  if (mSimulation)
+  {
+    // Publish joint states to /joint_states
+    sensor_msgs::JointState state;
+    state.header.stamp = ros::Time::now();
```
Comment (amalnanavati) [inline src/Ada.cpp:375]: Does the joint state message need a frame of reference?

**Review by amalnanavati — COMMENTED** [review]
Everything looks good. Once we resolve my comment/question about whether the JointState message needs a frame of reference, it should be good to go.

File: src/Ada.cpp (hunk 1 above)
Comment (egordon) [reply src/Ada.cpp:375]: I think not. The joint state is reference-frame invariant, and the official documentation only specifies the header timestamp (https://docs.ros.org/en/melodic/api/sensor_msgs/html/msg/JointState.html).

Comment (amalnanavati) [reply src/Ada.cpp:375]: Oh I see -- [the robot state publisher publishes a TF frame per joint name](http://wiki.ros.org/robot_state_publisher/Tutorials/Using%20the%20robot%20state%20publisher%20on%20your%20own%20robot). So this message is in fact what specifies the frames.

**Review by amalnanavati — APPROVED** [review]
LGTM

**Pull request merged**